src/bin/cargo/cli.rs

```diff
@@ -359,9 +356,11 @@ For more information, see issue #12207 <https://github.com/rust-lang/cargo/issue
                 let global_args = GlobalArgs::new(sub_args);
                 let new_args = cli(gctx).no_binary_name(true).try_get_matches_from(alias)?;
 
-                let new_cmd = match new_args.subcommand_name() {
-                    Some(new_cmd) => new_cmd,
-                    None => return Err(anyhow!("subcommand is required, add a subcommand to the command alias {cmd} .").into()),
+                let Some(new_cmd) = new_args.subcommand_name() else {
```

Comment:
I don't think we need an additional commit switching from match to let-else. Please squash the change into the same commit.

Also, <cbd5bbbb40bffca78b2590dd480593eeefba922c> is not an atomic commit. It basically did three things: remove `println!`, switch to let-else, and add a test. This is not ideal and would appreciate if you could clean them up.

Reply:
I'm very sorry for adding to your work burden. I didn't check it clearly. It's my problem.

I unconditionally obey your review opinions and welcome any suggestions and guidance. I hope Cargo will become better.

I hope my contribution is helpful and not annoying to you. grateful.